src/lib.rs

```diff
@@ -446,28 +496,52 @@ impl PyVM {
             RunEnterResult::Executed(NonEmptyValue::Failure(f)) => {
                 PyFailure::from(f).into_py(py).into_bound(py).into_any()
             }
-            RunEnterResult::NotExecuted => PyNone::get_bound(py).to_owned().into_any(),
+            RunEnterResult::NotExecuted(_retry_info) => PyNone::get_bound(py).to_owned().into_any(),
         })
     }
 
     fn sys_run_exit_success(
         mut self_: PyRefMut<'_, Self>,
         buffer: &Bound<'_, PyBytes>,
+    ) -> Result<PyAsyncResultHandle, PyVMError> {
+        CoreVM::sys_run_exit(
+            &mut self_.vm,
+            RunExitResult::Success(buffer.as_bytes().to_vec().into()),
+            RetryPolicy::None,
+        )
+        .map(Into::into)
+        .map_err(Into::into)
+    }
+
+    fn sys_run_exit_failure(
```

Comment:
Perhaps to avoid any confusion is better to rename this to `sys_run_exit_failure_terminal`